src/screenshot/sentry_screenshot_windows.c

```diff
@@ -0,0 +1,152 @@
+#include "sentry_screenshot.h"
+
+#include "sentry_logger.h"
+#include "sentry_path.h"
+
+#ifndef DWMWA_EXTENDED_FRAME_BOUNDS
+#    define DWMWA_EXTENDED_FRAME_BOUNDS 9
+#endif
+
+typedef enum { GpOk = 0 } GpStatus;
+typedef struct {
+    UINT32 GdiplusVersion;
+    PVOID DebugEventCallback;
+    BOOL SuppressBackgroundThread;
+    BOOL SuppressExternalCodecs;
+} GdiplusStartupInput;
+typedef void GpBitmap;
+typedef void GpImage;
+
+typedef GpStatus(WINAPI *GdiplusStartup_t)(
+    ULONG_PTR *token, const GdiplusStartupInput *input, void *output);
+typedef GpStatus(WINAPI *GdipCreateBitmapFromHBITMAP_t)(
+    HBITMAP hbm, HPALETTE hpal, GpBitmap **bitmap);
+typedef GpStatus(WINAPI *GdipSaveImageToFile_t)(GpImage *image,
+    const wchar_t *filename, const CLSID *encoder, const void *params);
+typedef GpStatus(WINAPI *GdipDisposeImage_t)(GpImage *image);
+
+typedef HRESULT(WINAPI *DwmGetWindowAttribute_t)(
+    HWND hwnd, DWORD dwAttribute, PVOID pvAttribute, DWORD cbAttribute);
+
+static bool
+save_bitmap(HBITMAP bitmap, const wchar_t *path)
+{
+    HMODULE gdiplus = LoadLibraryW(L"gdiplus.dll");
+    if (!gdiplus) {
+        return false;
+    }
+
+    GdiplusStartup_t pGdiplusStartup
+        = (GdiplusStartup_t)GetProcAddress(gdiplus, "GdiplusStartup");
+    GdipCreateBitmapFromHBITMAP_t pGdipCreateBitmapFromHBITMAP
+        = (GdipCreateBitmapFromHBITMAP_t)GetProcAddress(
+            gdiplus, "GdipCreateBitmapFromHBITMAP");
+    GdipSaveImageToFile_t pGdipSaveImageToFile
+        = (GdipSaveImageToFile_t)GetProcAddress(gdiplus, "GdipSaveImageToFile");
+    GdipDisposeImage_t pGdipDisposeImage
+        = (GdipDisposeImage_t)GetProcAddress(gdiplus, "GdipDisposeImage");
+
+    if (!pGdiplusStartup || !pGdipCreateBitmapFromHBITMAP
+        || !pGdipSaveImageToFile || !pGdipDisposeImage) {
+        FreeLibrary(gdiplus);
```

Comment:
Actually, all of these should log a warning, however slight the chance of failing.